packages/subsecond/subsecond-cli/src/main.rs

```diff
@@ -0,0 +1,678 @@
+use subsecond_cli_support::{create_jump_table, move_func_initiailizers};
+use target_lexicon::{Environment, Triple};
+use tokio::{
+    io::AsyncBufReadExt,
+    net::TcpListener,
+    process::{Child, Command},
+    time::Instant,
+};
+use tokio_tungstenite::WebSocketStream;
+use tracing::info;
+
+#[derive(Debug, Parser)]
+struct Args {
+    #[clap(long)]
+    target: Option<String>,
+}
+
+/// The main loop of the hotreload process
+///
+/// 1. Create initial "fat" build
+/// 2. Identify hotpoints from the incrementals. We ignore dependency hotpoints for now, but eventually might want to aggregate workspace deps together.
+/// 3. Wait for changes to the main.rs file
+/// 4. Perform a "fast" build
+/// 5. Diff the object files, walking relocations, preserving local statics
+/// 6. Create a minimal patch file to load into the process, including the changed symbol list
+/// 7. Pause the process with lldb, run the "hotfn_load_binary_patch" command and then continue
+/// 8. Repeat
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    // Go through the linker if we need to
+    if let Ok(action) = std::env::var("HOTRELOAD_LINK") {
+        return link(action).await;
+    }
+
+    tracing_subscriber::fmt::init();
+
+    let args = Args::parse();
+    let target: Triple = args
+        .target
+        .map(|t| t.parse().unwrap())
+        .unwrap_or_else(|| Triple::host());
+
+    // Save the state of the rust files
+    let src_folder = subsecond_folder().join("subsecond-harness/src/");
+    let main_rs = src_folder.join("main.rs");
+
+    // Modify the main.rs mtime so we skip "fresh" builds
+    // Basically `touch main.rs` in the directory
+    std::fs::File::open(&main_rs)?.set_modified(SystemTime::now())?;
+
+    // Perform the initial build
+    let epoch = SystemTime::UNIX_EPOCH;
+    let now = std::time::Instant::now();
+    tracing::debug!("Starting build for target {target:?}...");
+    let result = initial_build(&target).await?;
+    tracing::debug!(
+        "Initial build: {:?} -> {}",
+        now.elapsed(),
+        &result.output_location,
+    );
+
+    // copy the exe and give it a "fat" name. todo: wipe the ld entry that points to `/deps`
+    let exe = &result.output_location;
+    let fat_exe = exe.with_file_name(format!(
+        "fatharness-{}",
+        epoch.elapsed().unwrap().as_millis()
+    ));
+    std::fs::copy(&exe, &fat_exe).unwrap();
+
+    // Launch the fat exe. We'll overwrite the slim exe location, so this prevents the app from bugging out
+    let app = launch_app(&fat_exe, &target)?;
+
+    // Wait for the websocket to come up
+    let mut client = wait_for_ws(9393, &target).await?.unwrap();
+    tracing::info!("Client connected");
+
+    // Watch the source folder for changes
+    let mut watcher = FsWatcher::watch(src_folder)?;
+
+    while let Some(Ok(event)) = watcher.rx.next().await {
+        if event.kind != notify::EventKind::Modify(ModifyKind::Data(DataChange::Content)) {
```

Comment:
Can confirm that wasm now works! Only needed to lessen the restriction for the modify event.

```suggestion
        if !matches!(event.kind, notify::EventKind::Modify(_)) {
```